libs/langchain-openai/src/azure/chat_models_sdk.ts

```diff
@@ -0,0 +1,391 @@
+  azureOpenAIEndpoint?: string;
+
+  azureOpenAIApiKey?: string;
+
+  azureOpenAIApiCompletionsDeploymentName?: string;
+
+  private client: AzureOpenAIClient;
+
+  constructor(
+    fields?: Partial<OpenAIChatInput> &
+      Partial<AzureOpenAIInput> &
+      BaseLLMParams & {
+        configuration?: AzureOpenAIClientOptions;
+      }
+  ) {
+    super(fields ?? {});
+
+    this.azureOpenAIEndpoint =
+      fields?.azureOpenAIEndpoint ??
+      getEnvironmentVariable("AZURE_OPENAI_API_ENDPOINT");
+
+    this.azureOpenAIApiCompletionsDeploymentName =
+      fields?.azureOpenAIApiDeploymentName ??
+      getEnvironmentVariable("AZURE_OPENAI_API_DEPLOYMENT_NAME");
+
+    this.azureOpenAIApiKey =
+      fields?.azureOpenAIApiKey ??
+      getEnvironmentVariable("AZURE_OPENAI_API_KEY");
+
+    if (!this.azureOpenAIApiKey) {
+      throw new Error("Azure OpenAI API key not found");
+    }
+
+    if (!this.azureOpenAIEndpoint) {
+      throw new Error("Azure OpenAI Endpoint not found");
+    }
+
+    if (!this.azureOpenAIApiCompletionsDeploymentName) {
+      throw new Error("Azure OpenAI Completion Deployment name not found");
+    }
+
+    this.modelName = fields?.modelName ?? this.modelName;
+    this.prefixMessages = fields?.prefixMessages ?? this.prefixMessages;
+    this.modelKwargs = fields?.modelKwargs ?? {};
+    this.timeout = fields?.timeout;
+
+    this.temperature = fields?.temperature ?? this.temperature;
+    this.topP = fields?.topP ?? this.topP;
+    this.frequencyPenalty = fields?.frequencyPenalty ?? this.frequencyPenalty;
+    this.presencePenalty = fields?.presencePenalty ?? this.presencePenalty;
+    this.n = fields?.n ?? this.n;
+    this.logitBias = fields?.logitBias;
+    this.maxTokens = fields?.maxTokens;
+    this.stop = fields?.stop;
+    this.user = fields?.user;
+    this.functions = fields?.functions;
+    this.functionCall = fields?.functionCall;
+    this.azureExtensionOptions = fields?.azureExtensionOptions;
+
+    this.streaming = fields?.streaming ?? false;
+
+    if (this.n > 1) {
+      throw new Error(
+        "Cannot use n > 1 in OpenAIChat LLM. Use ChatOpenAI Chat Model instead."
+      );
+    }
+
+    if (this.azureOpenAIApiKey) {
+      if (!this.azureOpenAIApiCompletionsDeploymentName) {
+        throw new Error("Azure OpenAI API deployment name not found");
+      }
+      this.azureOpenAIApiKey = this.azureOpenAIApiKey ?? "";
+    }
+
+    const azureKeyCredential: AzureKeyCredential = new AzureKeyCredential(
+      this.azureOpenAIApiKey
+    );
+    this.client = new AzureOpenAIClient(
+      this.azureOpenAIEndpoint ?? "",
+      azureKeyCredential,
+      {
+        additionalPolicies: [
+          {
+            policy: {
+              name: "disableKeepAlive",
+              sendRequest: (request, next) => {
+                request.disableKeepAlive = true;
+                return next(request);
+              },
+            },
+            position: "perCall",
+          },
+        ],
+      }
```

Comment:
Why is this needed?

Reply:
For this issue - https://github.com/Azure/azure-sdk-for-js/issues/27959